### ISSUE TITLE
Incomplete documentation aws-sdk-js-v3

### ISSUE THREAD

**Issue by MeRahulAhire** [aws/aws-sdk-js-v3#2184]
Recently I'm to get a hands on demo on aws-sdk v3 but the problem is that there's lots of things in the docs that are missing. I want to know when will the Aws SDK JavaScript V3 docs will be illustrative and completed like the older V2 docs with better UI and search bar. It seems to refer the older docs and the syntax of newer version SDK is different. 

There's no mention of document client in that docs how to use other API. Same is with S3. There's no options for presigned url.

In short I want to know why its left incomplete

**Comment by ajredniwja** [comment]
Hi @MeRahulAhire thanks for opening this issue, I totally agree with you, the team is aware of it as well and is working on making the documentation better. Right now I use V2 docs for params and use it with V3 syntax.

To help you with some of the service that you mentioned, API reference for `documentClient` can be found [here](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/classes/_aws_sdk_lib_dynamodb.dynamodbdocument-1.html) with example shown in https://github.com/aws/aws-sdk-js-v3/pull/2097, for S3 example see [this](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/s3-examples.html) and how to generate a `presignedUrl`: example can be found [here](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_s3_request_presigner.html).

There are also some complete working examples that can be found [here](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/sdk-code-samples.html) for some of the most used services.

While the team works on making the documentation better, feel free to reach out if you have any other questions.

**Comment by MeRahulAhire** [comment]
I want to know why presigned URL isn't bundled with `client-S3` instead made a totally different component.
Last day I tried the `listbucketcommand` and it gives me the null object `[]` message and not actual results
Especially my complain would be why isn't everything centralised together at one place. 
It'd be I'd you specified the brief summary of  issues and bugs upfront in the new docs itself
The docs is somewhere the theory is other webpage and there's no proper navigation and synchronization between knowing theory to writing code at just one docs. Better example would be off reactjs, it's docs is really good..

**Comment by ajredniwja** [comment]
@MeRahulAhire, thanks for your feedback there, we are trying to make docs better and your feedback is really valuable. 
As for the `presigned-url` we have this blogpost [here](https://aws.amazon.com/blogs/developer/generate-presigned-url-modular-aws-sdk-javascript/) which explains everything. Its basically so the customers dont have to differentiate between `getSignedUrl()` and `getSignedUrlPromise()` any more.

**Comment by RanVaknin** [comment]
Hi @MeRahulAhire 

Sorry for the late response. We are aware of this issue, and are trying to rebuild our v3 docs to be more useful.
Im going to close this issue in favor of tracking it in the https://github.com/aws/aws-sdk-js-v3/issues/4318 here.

If this is still a concern for you, please upvote that issue so we can have more visibility to the community impact.

Thank you very much,
Ran~
